packages/hash/api/src/graphql/resolvers/entityType/entityTypeTypeFields.ts

```diff
@@ -36,30 +57,30 @@ const entityType: Resolver<
   if (gqlEntityType.entityType) {
     return gqlEntityType.entityType;
   }
-  return getEntityTypeType(dataSources).then((type) => type.toGQLEntityType());
+  return getEntityTypeById(dataSources, gqlEntityType.entityId);
 };
 
 /**
  * Get the entityTypeId of an EntityType, i.e. the entityId of the "EntityType" EntityType.
  */
-const entityTypeId: Resolver<
+const entityTypeIdResolver: Resolver<
```

Comment:
I have made changes to these resolvers. As I understand it this had something to do with the system `EntityType` type before, but after testing with a graphQL query, the result of resolving `entityTypeId` was incorrect, always returning that of the system `EntityType`. The changes to this - and the next couple of resolver - might be unnecessary, but their former behaviour seems to be incorrect. 

I would like to know if this should in fact still resolve to the system `EntityType` or not :)

Reply:
Yes, this should still resolve to the system `EntityType`. It's confusing and I welcome suggestions how to make it less so, but:

The short version is that for **both** an `Entity` and an `EntityType`, the `entityId` property is the fixed id of _that thing_, whereas the `entityTypeId` is the fixed id of the _type it belongs to_.

**entities**
1. we have things called "entities" which are instances of a particular `EntityType`
2. entities have an `entityId` which is _their own_ fixed id
3. entities also have an `entityTypeId` which is the fixed id of the instance of `EntityType` they implement

**entity types**
1. entity types are also instances of a particular `EntityType` - the `EntityType` called "EntityType"
2. therefore, their `entityTypeId` is the fixed id of the "EntityType" `EntityType`
3. the fixed id of any given `EntityType` is its `entityId` (the same for entities)

Reply:
Gotcha, that is understandable. That means i should be using the `entityId` only for referring to the id of that "thing"s `entityTypeId`. Otherwise querying `entityTypeId` would result in system `EntityType`. 

I'll revert to the setup to a version that had the two new resolvers in another location.

Reply:
Cool, thank you.

I'm not 100% sold on how it is now btw - originally we wanted to have entity types as just like any other entity -which is why unifying the concept of `entityId` and `entityTypeId` across them - but they need treatment in various ways which means we split them out in the db and API schema.

The spec is also slightly different again, as it says that blocks will receive EntityTypes as a JSON schema object with an additional `entityTypeId` property, which represents its fixed id.

Reply:
Created [(internal) asana task](https://app.asana.com/0/0/1201686359879649/f) about this